packages/eslint-plugin-pf-codemods/src/rules/v6/emptyStateHeaderMoveIntoEmptyState/emptyStateHeader-move-into-emptyState.ts

```diff
@@ -129,8 +130,48 @@ module.exports = {
 
         const iconPropValue = getExpression(headerIconAttribute?.value);
 
-        const emptyStateIconComponent =
-          iconPropValue?.type === "JSXElement" ? iconPropValue : undefined;
+        const iconElementIdentifier =
+          iconPropValue?.type === "JSXElement" &&
+          iconPropValue.openingElement.name.type === "JSXIdentifier"
+            ? iconPropValue.openingElement.name
+            : undefined;
+
+        const iconPropIsEmptyStateIconComponent = () => {
+          const emptyStateIconImport = imports.find(
+            (specifier) => specifier.imported.name === "EmptyStateIcon"
+          );
+
+          return (
+            iconElementIdentifier?.name === emptyStateIconImport?.local.name
+          );
+        };
+
+        const iconPropIsIconElement = () => {
+          const pfIconsPackage = "@patternfly/react-icons";
+          const { imports: iconSpecifiers } = getFromPackage(
+            context,
+            pfIconsPackage
+          );
+          const iconDefaultSpecifiers = getDefaultImportsFromPackage(
+            context,
+            pfIconsPackage
+          );
+          const allIconSpecifiers = [
+            ...iconSpecifiers,
+            ...iconDefaultSpecifiers,
+          ];
+
+          return (
+            iconElementIdentifier !== undefined &&
+            allIconSpecifiers.some(
+              (spec) => spec.local.name === iconElementIdentifier.name
+            )
+          );
+        };
+
+        const emptyStateIconComponent = iconPropIsEmptyStateIconComponent()
+          ? (iconPropValue as JSXElement)
```

Comment:
Is this type assertion definitely valid?

Reply:
Yes, it is quite indirect, but the `iconPropIsEmptyStateIconComponent` function also checks that `iconElementIdentifier` is not undefined, which is true when `iconPropValue?.type === "JSXElement"`.

Reply:
What if there is no `EmptyStateIcon` import? Would that cause `emptyStateIconImport` to be `undefined` and `iconPropIsEmptyStateIconComponent` to incorrectly return true (since `undefined === undefined` is true)?

Reply:
Oh, good catch, that could cause some trouble, thanks for finding that!

Reply:
That's what I'm here for! 🙂